src/main/kotlin/com/exactpro/th2/common/utils/event/MessageUtils.kt

```diff
@@ -6,29 +6,68 @@ import com.exactpro.th2.common.grpc.Message
 import com.exactpro.th2.common.grpc.MessageGroup
 import com.exactpro.th2.common.grpc.MessageGroupOrBuilder
 import com.exactpro.th2.common.grpc.RawMessage
+import com.google.protobuf.TextFormat
 
 fun Message.toGroup(): MessageGroup = MessageGroup.newBuilder().add(this).build()
 fun RawMessage.toGroup(): MessageGroup = MessageGroup.newBuilder().add(this).build()
 fun AnyMessage.toGroup(): MessageGroup = MessageGroup.newBuilder().addMessages(this).build()
 
-val AnyMessageOrBuilder.sessionAlias: String
+val AnyMessageOrBuilder.sessionAlias: String?
     get() = when {
-        hasMessage() -> message.metadata.id.connectionId.sessionAlias
-        hasRawMessage() -> message.metadata.id.connectionId.sessionAlias
+        hasMessage() -> message.sessionAlias
+        hasRawMessage() -> rawMessage.sessionAlias
         else -> error("Unsupported message kind: $kindCase")
     }
 
-fun MessageGroup.Builder.add(message: Message): MessageGroup.Builder = apply { addMessagesBuilder().message = message }
-fun MessageGroup.Builder.add(message: RawMessage): MessageGroup.Builder = apply { addMessagesBuilder().rawMessage = message }
+val Message.sessionAlias: String?
+    get() = sessionAliasOrEmpty.run {
+        when {
+            isEmpty() -> null
+            else -> this
+        }
+    }
+
+val RawMessage.sessionAlias: String?
+    get() = sessionAliasOrEmpty.run {
+        when {
+            isEmpty() -> null
+            else -> this
+        }
+    }
 
 val MessageGroupOrBuilder.sessionAlias: String?
+    get() = sessionAliasOrEmpty.run {
+        when {
+            isEmpty() -> null
+            else -> this
+        }
+    }
+
+val AnyMessageOrBuilder.sessionAliasOrEmpty: String
+    get() = when {
+        hasMessage() -> message.sessionAliasOrEmpty
+        hasRawMessage() -> rawMessage.sessionAliasOrEmpty
+        else -> error("Unsupported message kind: $kindCase")
+    }
+
+val Message.sessionAliasOrEmpty: String
+    get() = metadata.id.connectionId.sessionAlias
+
+val RawMessage.sessionAliasOrEmpty: String
+    get() = metadata.id.connectionId.sessionAlias
```

Comment:
I'm still in doubt of need for 2 separate methods for retrieving session alias, but if we really need them - these ones should be called just `sessionAlias` and the ones returning `null` - `sessionAliasOrNull`